delira/training/predictor.py

```diff
@@ -553,3 +609,83 @@ def calc_metrics(batch: LookupConfig, metrics=None, metric_keys=None):
+
+    @staticmethod
+    def tta(transforms: tuple = (), reduce_fn=None, inverse_transforms=()):
+        """
+        Decorator function to apply test-time augmentation during prediction
+
+        Parameters
+        ----------
+        transforms : tuple
+            a tuple of transforms to call on the ``data_dict`` for test-time
+            augmentation. Each transform will be executed separately on a
+            new data_dict.
+        reduce_fn :
+            function to reduce the ``tta_results``. The reduction should
+            only be applied along axis 0, since the other dimensions must be
+            retained (these are the output dimensions).
+
+        inverse_transforms : tuple
+            transforms to apply, if the transform has to be reverted before
+            reducing (e.g. in Segmentation tasks)
+
+        Returns
+        -------
+        function
+            the decorated function
+
+        """
+
+        # use mean as default reduce function
+        if reduce_fn is None:
+            reduce_fn = partial(np.mean, axis=0)
+
+        # add no transformation to transforms to also predict original data
+        transforms = (None, *transforms)
+
+        # check whether to transform back and add None to inverse transforms
+        # if necessary
+        if inverse_transforms:
+            transform_back = True
+            inverse_transforms = (None, *inverse_transforms)
+        else:
+            transform_back = False
+
+        # decorates the actual function
+        def decorate_fn(function):
+            # wraps the actual function by iterating over the transforms
+            def wrapper(data_dict, **kwargs):
+                results = {}
+                for idx, trafo in enumerate(transforms):
+                    # apply transforms if possible
+
+                    curr_data = copy.deepcopy(data_dict)
+                    if trafo is not None:
+                        curr_data = trafo(**curr_data)
+
+                    _result = function(curr_data, **kwargs)
+
+                    # check whether to transform back
+                    if transform_back:
+                        _inv_trafo = inverse_transforms[idx]
+                        # check if trafo must be applied
+                        if _inv_trafo is not None:
+                            _result = _inv_trafo(**_result)
+
+                    # add each item in current result dict to total results
+                    # dict
+                    for k, v in flatten_dict(_result).items():
+                        if k in results:
+                            results[k] += [v]
+                        else:
+                            results[k] = [v]
+
+                # convert every value list inside the results dict to arrays
+                # and apply the reduce function afterwards.
+                for k, v in results.items():
+                    results[k] = reduce_fn(np.array(v))
```

Comment:
I think this would be to complicated in this case, since this is a good standard value and usually there is only one prediction, that should be reduced. Also one could do this via the transforms